scalability_and_performance/understanding-node-observability-operator.adoc

```diff
@@ -0,0 +1,363 @@
+
+.Prerequisites
+* The Node Observability Operator has been installed.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+. Log in to the {product-title} CLI as a user with the `cluster-admin` role:
++
+[source,terminal]
+----
+$ oc login -u kubeadmin https://<HOSTNAME>:6443
+----
+
+. Create a new project named `node-observability-operator`:
++
+[source,terminal]
+----
+$ oc project node-observability-operator
+----
+
+. Create a CR file named `nodeobservability.yaml` that contains the following text:
++
+[source,yaml]
+----
+    apiVersion: nodeobservability.olm.openshift.io/v1alpha1
+    kind: NodeObservability
+    metadata:
+      name: cluster <1>
+    spec:
+      labels:
+      type: crio-kubelet
+----
+<1> You must specify the name as `cluster` because it is applied cluster-wide.
+
+. Run the `NodeObservability` CR:
++
+[source,terminal]
+----
+oc apply -f nodeobservability.yaml
+----
+
++
+.Example output
+[source,terminal]
+----
+nodeobservability.olm.openshift.io/cluster created
+----
+
+. Review the status of the `NodeObservability` CR:
++
+[source,terminal]
+----
+$ oc get nob/cluster -o yaml
+----
+
++
+.Example output
+[source,terminal]
+----
+apiVersion: v1
+items:
+- apiVersion: nodeobservability.olm.openshift.io/v1alpha1
+  kind: NodeObservability
+  metadata:
+    annotations:
+      kubectl.kubernetes.io/last-applied-configuration: |
+        {"apiVersion":"nodeobservability.olm.openshift.io/v1alpha1","kind":"NodeObservability","metadata":{"annotations":{},"name":"cluster"},"spec":{"labels":null,"type":"crio-kubelet"}}
+    creationTimestamp: "2022-07-05T07:21:59Z"
+    finalizers:
+    - NodeObservability
+    generation: 1
+    name: cluster
+    resourceVersion: "131690"
+    uid: 897815d9-929d-48d9-9f1a-53f1390b456c
+  spec:
+    type: crio-kubelet
+  status:
+    conditions:
+      conditions:
+      - lastTransitionTime: "2022-07-05T07:33:54Z"
+        message: 'DaemonSet node-observability-ds ready: true NodeObservabilityMachineConfig
+          ready: true'
+        reason: Ready
+        status: "True"
+        type: Ready
+    count: 1
+    lastUpdated: "2022-07-05T07:33:55Z"
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+----
+
++
+`NodeObservability` CR run is completed when the reason `Ready` and status `True`.
+
+
+== Running profiling queries
+
+Profiling query is a blocking operation that fetches CRI-O and Kubelet profiling for a duration of 30 seconds. To request a profiling query, you have to create a `NodeObservabilityRun` resource.
+
+[IMPORTANT]
+====
+You can request only one profiling query at any point of time.
+====
+
+.Prerequisites
+* The Node Observability Operator has been installed.
+* The `NodeObservability` custom resource (CR) has been created.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a `NodeObservabilityRun` resource file named `nodeobservabilityrun.yaml` that contains the following text:
```

Comment:
@alebedev87 Did you mean 

> . Create the NodeObservabilityRun CR namespace:
> 
> $ oc project node-observability-operator

Reply:
@alebedev87 Sherine told me to remove `$ oc project node-observability-operator` step. She thinks there's no need to recreate project as the user would have done it at the previous step while installing the operator.

@sherine-k Could you please pitch in and share your thoughts.

Reply:
`oc project` command doesn't create a project, it switches to the the specified project. You have to switch to it because all the commands for `NodeObservabilityRun` have to be done in the project or changed to add the namespace.

Reply:
Hi I agree we should keep `oc project node-observability-operator`.
This command switches to the existing namespace `node-observability-operator`.
My comment was more to say that we don't have to create the namespace again. I must have not read properly the command. So sorry about that

Reply:
Fixed and pushed the changes.

Reply:
@alebedev87 Do you think example output is needed for this command ? If yes, please share.

Reply:
No, just `oc project node-observability-operator` command is enough.